src/lib/TSL/Splitter.hs

```diff
@@ -126,7 +126,7 @@ split spec@Specification{assumptions, guarantees} =
     splitGuars = splitFormulas guarantees connComp
     splitAssumptions  = splitFormulas boundAssumptions connComp
   in
-    fmap cleanSymboltable $ buildSpecs $ addFreeAssumptions freeAssumptions $ zip splitAssumptions splitGuars
+    fmap cleanSymboltable $ buildSpecs $ addFreeAssumptions freeAssumptions $ filter (not . null . snd) $ zip (splitAssumptions ++ repeat []) splitGuars
```

Comment:
How comes that |splitAssumptions| /= |splitGuars| is possible now?

Reply:
Both lists have as many elements as connected components in the dependency graph plus all formulas that do not match any component. These could be e.g. `true` or `false` as they do not depend on any variable. For guarantees we may not ignore a `false`. For assumption, however we can ignore a `false` as we assume unrealizability of `assumption -> false`.